liquid-base/src/Data/Bits.hs

```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE Trustworthy #-}
```

Comment:
I am surprised we need to add the pragma for anything except `Prelude`. For this and the other modules GHC should just transitively infer. But it looks like that wasn't the case?

Reply:
First, I tried to insert pragma only in Prelude but GHC still complained about `safe import Data.Bits` and `safe import Data.Word` in the project I was building. It didn't compile until I inserted pragmas in all liquid modules that the project uses.

Should I also file an issue to GHC about not inferring safety?

Reply:
Uhm, maybe this is actually expected after all? I was just reading the [Safe Haskell Inference](https://downloads.haskell.org/~ghc/7.8.4/docs/html/users_guide/safe-haskell.html#safe-inference) section in GHC's documentation, and it says:

> "[..]In the case where a module is compiled without one of -XSafe, -XTrustworthy or -XUnsafe being used, GHC will try to figure out itself if the module can be considered safe. This safety inference will never mark a module as trustworthy, only as either unsafe or as safe. GHC uses a simple method to determine this for a module M: If M would compile without error under the -XSafe flag, then M is marked as safe. If M would fail to compile under the -XSafe flag, then it is marked as unsafe.[...]"

So perhaps the extra pragmas were necessary, although it's a bit unsatisfactory. I am also noticing that the original `base` modules have different inferred degree of safety: `Data.Bits` and `Prelude` for example are marked `Trustworthy`, but `Data.Word` is marked as `Safe`, so perhaps this has something to do with the chain of pragmas you had to add. Perhaps @kosmikus can shed some extra light here?